### PR TITLE
Support ethermint

### LINKED ISSUE #145: orchestrator is not compatible with ethermint
Add a cargo feature to switch to ethermint mode:
- address derivation method
- EthAccount decoding
- default coin type change to 60

something similar to https://github.com/informalsystems/ibc-rs/pull/1295

### PR DESCRIPTION
**Pull request by yihuang**
Closes #145

Add a cargo feature to switch to ethermint mode:
- address derivation method
- EthAccount decoding
- default coin type change to 60
- change message hashing in signing

deep_space PR: https://github.com/iqlusioninc/deep_space/pull/1

### CODE REVIEW

File: orchestrator/client/src/main.rs [hunk 1]
```diff
@@ -123,8 +126,13 @@ async fn main() {
         // todo actually query metadata for this
         let is_cosmos_originated = !gravity_denom.starts_with("gravity");
         let amount = args.flag_amount.unwrap().parse().unwrap();
-        let cosmos_key = CosmosPrivateKey::from_phrase(&args.flag_cosmos_phrase, "")
-            .expect("Failed to parse cosmos key phrase, does it have a password?");
+        let hd_path = args
+            .flag_hd_wallet_path
+            .as_deref()
+            .unwrap_or("m/44'/118'/0'/0/0");
```
[the remark this answers is not in the document]
Comment (yihuang) [reply orchestrator/client/src/main.rs]: done, moved `DEFAULT_HD_PATH` to `cosmos_gravity::crypto::`.

File: orchestrator/register_delegate_keys/src/main.rs [hunk 2]
```diff
@@ -79,16 +82,22 @@ async fn main() {
     let contact = connections.contact.unwrap();
     wait_for_cosmos_node_ready(&contact).await;
 
-    let validator_key = CosmosPrivateKey::from_phrase(&args.flag_validator_phrase, "")
-        .expect("Failed to parse validator key");
+    let hd_path = args
+        .flag_hd_wallet_path
+        .as_deref()
+        .unwrap_or("m/44'/118'/0'/0/0");
```
[the remark this answers is not in the document]
Comment (yihuang) [reply orchestrator/register_delegate_keys/src/main.rs]: done

File: orchestrator/Cargo.toml [hunk 3]
```diff
@@ -13,3 +13,5 @@ members = [
 	"register_delegate_keys",
 	"gorc",
 ]
+[patch.crates-io]
+deep_space = { git = "https://github.com/yihuang/deep_space/", branch = "ethermint" }
```
[the remark this answers is not in the document]
Comment (yihuang) [reply orchestrator/Cargo.toml:17]: > @yihuang I am going to try and merge your changes to [deep space ](https://github.com/yihuang/deep_space/) in with the iqlusioninc deep space changes so we can just depend on that upstream repo.

how's that going? 😄

File: orchestrator/cosmos_gravity/src/crypto.rs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    private_key::PrivateKey as InnerPrivateKey,
+    Address, MessageArgs, Msg,
+};
+
+#[cfg(feature = "ethermint")]
+pub const DEFAULT_HD_PATH: &str = "m/44'/60'/0'/0/0";
+#[cfg(not(feature = "ethermint"))]
+pub const DEFAULT_HD_PATH: &str = "m/44'/118'/0'/0/0";
+
+/// PriateKey wraps cosmos private key, switch between cosmos and ethermint behavior according to compile feature.
```
Comment (JayT106) [inline orchestrator/cosmos_gravity/src/crypto.rs:14]: `PriateKey` -> `PrivateKey`

Comment (yihuang) [reply orchestrator/cosmos_gravity/src/crypto.rs]: good catch, fixed

**Comment by yihuang** [comment]
fixed conflicts and squashed.

**Comment by thomas-nguy** [comment]
can be close https://github.com/crypto-org-chain/gravity-bridge/pull/31

**Pull request closed without merging**